### DISCUSSION TITLE
Jacobian invert blocks for generalised behaviours

### DISCUSSION THREAD

**Discussion by mhpierre** [thelfer/tfel#357] [Q&A] [answered]
I am trying to implement a coupled multiphysics model with plasticity in MFront. It has been a great experience so far but I have a small issue with getting the Jacobian invert to calculate the tangent operator. I have been referring to this page, part 3.2 : https://tfel.sourceforge.net/implicit-dsl.html

I am trying to get the derivative of the elastic strain with respect to the total strain. I have not been able to get the invert block through the `iJ_fε_el_ε_el` as I get an error saying that it is not defined, am I missing some keyword there ?

I also tried using the `getIntegrationVariablesDerivatives_ε_tot` function, but it seems this function outputs a second order tensor for `∂Δε_el∕∂Δε_tot`, where I would expect this derivative to be a fourth order tensor, if I am not mistaken ?

I give the model code attached
Thank you very much in advance,

Maxime
[TM-acc.mfront.txt](https://github.com/thelfer/tfel/files/10391505/TM-acc.mfront.txt)

**Comment by thelfer** [comment, marked as answer]
@LordKthulhu (love your alias),

the  `getIntegrationVariablesDerivatives_ε_tot` has various overloads which depends on the order of declaration of the internal state variables. Hence, the correct code in your case is:

```c++
    Stensor ∂S∕∂Δε_tot ;
    Stensor4 ∂Δε_el∕∂Δε_tot ;
    getIntegrationVariablesDerivatives_ε_tot(∂S∕∂Δε_tot, ∂Δε_el∕∂Δε_tot);
```
